lib/templates-sync.php

```diff
@@ -5,6 +5,18 @@
  * @package gutenberg
  */
 
+/**
+ * Render a Block Based template and returns its output.
+ *
+ * @access private
+ * @internal
+ *
+ * @param string $template_path Template file path.
+ */
+function _gutenberg_read_template( $template_path ) {
+	return shell_exec( 'php ' . __DIR__ . '/render-script.php ' . $template_path );
```

Comment:
We should ensure that the new process uses the current PHP interpreter, not just the system's default `php`. We can do this with the [`PHP_BINARY` constant as of 5.4](https://www.php.net/manual/en/reserved.constants.php):

```suggestion
	return shell_exec( PHP_BINARY . ' ' . __DIR__ . '/render-script.php ' . $template_path );
```

Reply:
I have ***deep*** misgivings about this, it's encouraging developers to bootstrap WordPress to gain access to WP API functions, making these files extremely fragile and easy to break. This also means a big performance hit.

There are also servers that don't allow `shell_exec` for security reasons

Reply:
I had a thought, aside from a whitelist, it would be good to add 2 extra checks:
 - A check to confirm this is being ran directly rather than via a browser request
 - A nonce to confirm that it is WordPress running the file and not some other piece of code

It may also be worth looking into using an additional parameter to override ini values and prevent the use of networking and file writing functions